src/cmd/api/main.go

```diff
@@ -84,8 +84,22 @@ func main() {
 		log.Fatal("Cannot create consul client : ", err)
 	}
 
-	// Register service on consul (only for local env)
-	registerService(cfg, discoveryClient)
+	// Start service discovery
+	ctxDiscovery, cancelDiscovery := context.WithCancel(context.Background())
```

Comment:
Actually the good practice is to have one single ctx for all your subroutine within the main function.